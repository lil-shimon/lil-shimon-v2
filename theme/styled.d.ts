import "styled-components";

declare module "styled-components" {
  export interface DefaultTheme {
    mediaQuery: {
      sp: string;
      tablet: string;
      laptop: string;
    };
  }
}
