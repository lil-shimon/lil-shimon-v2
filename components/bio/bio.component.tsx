import * as Styles from "./bio.styles";

export const Bio = () => {
  return (
    <Styles.Container>
      <Styles.Title>Bio</Styles.Title>
      <Styles.BioContainer>
        <Styles.Year>2019</Styles.Year>
        <span>カナダ カルガリー留学</span>
      </Styles.BioContainer>
      <Styles.BioContainer>
        <Styles.Year>2021</Styles.Year>
        <span>日本大学 経済学部経済学科国際コース 卒業</span>
      </Styles.BioContainer>
      <Styles.BioContainer>
        <Styles.Year>2021</Styles.Year>
        <span>クェスタ株式会社 プログラマー 入社</span>
      </Styles.BioContainer>
      <Styles.BioContainer>
        <Styles.Year>2023-現在</Styles.Year>
        <span>フリーランスエンジニア</span>
      </Styles.BioContainer>
    </Styles.Container>
  );
};
