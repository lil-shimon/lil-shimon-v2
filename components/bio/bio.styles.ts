import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  gap: 1em;
`;

export const Title = styled.h1`
  padding-bottom: 0.5em;
  border-bottom: #333 1px dotted;
`;

export const BioContainer = styled.div`
  display: grid;
  grid-template-columns: 70px 4fr;
  align-items: center;
  gap: 1em;
`;

export const Year = styled.span`
  font-weight: bold;
`;
