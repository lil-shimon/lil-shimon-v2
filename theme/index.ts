import { DefaultTheme } from "styled-components";

const customMediaQuery = (minWidth: number) =>
  `@media (min-width: ${minWidth}px)`;

export const theme: DefaultTheme = {
  mediaQuery: {
    sp: customMediaQuery(576),
    tablet: customMediaQuery(768),
    laptop: customMediaQuery(992),
  },
};
