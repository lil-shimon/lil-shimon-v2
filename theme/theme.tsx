import { ReactNode } from "react";
import { ThemeProvider as StyledComponentThemeProvider } from "styled-components";

import { DefaultTheme } from "styled-components";

const customMediaQuery = (minWidth: number) =>
  `@media (min-width: ${minWidth}px)`;

const theme: DefaultTheme = {
  mediaQuery: {
    sp: customMediaQuery(576),
    tablet: customMediaQuery(768),
    laptop: customMediaQuery(992),
  },
};

type Props = {
  children: ReactNode;
};

export const ThemeProvider = ({ children }: Props) => {
  return (
    <StyledComponentThemeProvider theme={theme}>
      {children}
    </StyledComponentThemeProvider>
  );
};
