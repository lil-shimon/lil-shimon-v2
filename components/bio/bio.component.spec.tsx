import { render } from "@testing-library/react";
import { Bio } from "./bio.component";
import { ThemeProvider } from "@/theme";
import { expect } from "vitest";

describe("Bio", () => {
  it("should render", () => {
    const { asFragment } = render(<Bio />, {
      wrapper: ({ children }) => <ThemeProvider>{children}</ThemeProvider>,
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
