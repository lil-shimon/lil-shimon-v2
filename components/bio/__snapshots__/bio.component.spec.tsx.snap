// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bio > should render 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb kAByRm"
  >
    <h1
      class="sc-gtLWhw gImffP"
    >
      Bio
    </h1>
    <div
      class="sc-egkSDF hOJX"
    >
      <span
        class="sc-fAUdSK bbVLsX"
      >
        2019
      </span>
      <span>
        カナダ カルガリー留学
      </span>
    </div>
    <div
      class="sc-egkSDF hOJX"
    >
      <span
        class="sc-fAUdSK bbVLsX"
      >
        2021
      </span>
      <span>
        日本大学 経済学部経済学科国際コース 卒業
      </span>
    </div>
    <div
      class="sc-egkSDF hOJX"
    >
      <span
        class="sc-fAUdSK bbVLsX"
      >
        2021
      </span>
      <span>
        クェスタ株式会社 プログラマー 入社
      </span>
    </div>
    <div
      class="sc-egkSDF hOJX"
    >
      <span
        class="sc-fAUdSK bbVLsX"
      >
        2023-現在
      </span>
      <span>
        フリーランスエンジニア
      </span>
    </div>
  </div>
</DocumentFragment>
`;
